### ISSUE TITLE
MaterialDialog Descendants Fail w/ Dart SDK 1.22.0

### ISSUE THREAD

**Issue by mohammed90** [MikeMitterer/dart-material-design-lite#42]
The latest update to the Dart SDK from 1.21.1 to 1.22.0 breaks the functionality of MaterialAlertDialog, MaterialNotification, and MaterialSnackbar when compiled to JavaScript with Dart2JS compiler. It runs perfectly in Dartium, but it fails in Chrome and Firefox when compiled to JS with the following error:

`Uncaught Error: Unsupported operation: Cannot find class for: (String, {String okButton, String title}) => MaterialAlertDialog`

Here's the `main.dart` code to reproduce:
```Dart
// Copyright (c) 2017, Mohammed Al Sahaf. All rights reserved. Use of this source code
// is governed by a BSD-style license that can be found in the LICENSE file.

import 'dart:async';
import 'dart:html';

import 'package:console_log_handler/console_log_handler.dart';
import 'package:di/di.dart' as di;
import 'package:logging/logging.dart';
import 'package:mdl/mdl.dart';


@MdlComponentModel
@di.Injectable()
class App extends MaterialApplication {

	App() {
  }

	void run() {
		
		this._bind();
	}

	//- private -----------------------------------------------------------------------------------

	void _bind() {

    final MaterialButton button = MaterialButton.widget(querySelector("#alert-button"));
    final MaterialAlertDialog alertDialog = new MaterialAlertDialog();

		button.onClick.listen((MouseEvent event) {
        alertDialog("Alert Message").show();
		});

	}

}

Future main() async {
  configLogging();
	registerMdl();

	final MaterialApplication application = await componentFactory().rootContext(App).run(enableVisualDebugging: true);

	application.run();

}

void configLogging() {
	hierarchicalLoggingEnabled = true; // set this to true - its part of Logging SDK

	// now control the logging.
	// Turn off all logging first
	Logger.root.level = Level.ALL;
	Logger.root.onRecord.listen(new LogConsoleHandler());
}
```

And follows is the `index.html` content:
```HTML
<!DOCTYPE html>
<html>

	<head>
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<meta name="viewport" content="width=device-width, initial-scale=1.0">
		<title>MDL-Dart and Dart SDK 22 Bug</title>
		
		<link rel="stylesheet" href="packages/mdl/assets/styles/material.min.css">

		<link href='https://fonts.googleapis.com/css?family=Roboto:regular,bold,italic,thin,light,bolditalic,black,medium&amp;lang=en'
		    rel='stylesheet' type='text/css'>
		<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
		
		<link id="theme" rel="stylesheet" href="https://cdn.rawgit.com/MikeMitterer/dart-mdl-theme/v1.18.2/deep_purple-red/material.min.css">
	</head>

	<body>
		<div class="mdl-panel mdl-panel--with-heading mdl-panel--accent mdl-panel--light-text mdl-panel--shadow">
			<div class="mdl-panel__heading">Bug Demo</div>
			<div class="mdl-panel__content">
				<button id="alert-button" class="mdl-button mdl-button--raised mdl-button--primary mdl-ripple-effect">
					Alert
				</button>
			</div>
		</div>
		<script defer src="packages/browser/dart.js"></script>
		<script type="application/dart" src="main.dart" ></script>
	</body>

</html>
```

**Comment by MikeMitterer** [comment]
Hi @Mohammed90 - MANY Thanks for reporting!
Seems to be a major bug in dart2js compiler. I'm investigating. I'll be back as soon as I know more.

**Comment by MikeMitterer** [comment]
Filed a bug report: https://github.com/dart-lang/sdk/issues/28915

**Comment by MikeMitterer** [comment]
Solved with Dart 1.23.0
